common/shell-local/config.go

```diff
@@ -159,6 +162,24 @@ func Validate(config *Config) error {
 				fmt.Errorf("Bad script '%s': %s", path, err))
 		}
 	}
+
+	// Check for properly formatted go os types
+	supported_syslist := []string{"darwin", "freebsd", "linux", "openbsd", "solaris", "windows"}
+	if len(config.OnlyOn) > 0 {
+		for _, provided_os := range config.OnlyOn {
+			supported_os := false
+			for _, go_os := range supported_syslist {
+				if provided_os == go_os {
+					supported_os = true
+					break
+				}
+			}
+			if supported_os != true {
+				return fmt.Errorf("Invalid OS specified in only_on: '%s'", provided_os)
```

Comment:
Let's also add the list of valid runtimes to this error message, to help the user correct their template.

Reply:
Good idea! 👍 I wish there was an easy way to access the const in the runtime library :/

Reply:
Wording and spacing, aside this should cover this. Open to suggestions on wording and such as usual :)